### PR TITLE
input_osv3: validate plugins against a schema

### PR DESCRIPTION
**Pull request by mlangsdorf**
add a simple schema for plugins to schema and add a call to input_osv3
to validate the final plugins json against it.

Signed-off-by: Mark Langsdorf <mlangsdo@redhat.com>

### CODE REVIEW

File: atomic_reactor/schemas/user_params.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    "flatpak_base_image": {"type": "string"},
+    "git_branch": {"type": "string"},
+    "git_ref": {"type": "string"},
+    "git_uri": {"type": "string"},
+    "image_tag": {"type": "string"},
+    "imagestream_name": {"type": "string"},
+    "isolated": {"type": "boolean"},
+    "koji_parent_build": {"type": "string"},
+    "koji_target": {"type": "string"},
+    "koji_task_id": {"type": "string"},
```
Comment (lcarva) [inline atomic_reactor/schemas/user_params.json]: This is an integer.

**Comment by mlangsdorf** [comment]
Also, I tried to figure out how to use oneOf to constrain some of the user parameters that are exclusive to each other, but I couldn't figure out the syntax.

File: atomic_reactor/schemas/user_params.json (hunk 1 above)
Comment (mlangsdorf) [reply atomic_reactor/schemas/user_params.json]: Fixed.

File: atomic_reactor/plugins/input_osv3.py [hunk 2]
```diff
@@ -124,6 +124,8 @@ def remove_plugins_without_parameters(self):
             self.remove_plugin('exit_plugins', 'sendmail', 'no mailhost available')
         if not self.get_value('sources_command'):
             self.remove_plugin('prebuild_plugins', 'distgit_fetch_artefacts', 'no sources command')
+        # make sure the final json is valid
+        read_yaml(json.dumps(self.plugins_json), 'schemas/plugins.json')
```
[the remark this answers is not in the document]
Comment (mlangsdorf) [reply atomic_reactor/plugins/input_osv3.py]: Fixed.

File: atomic_reactor/schemas/plugins.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        },
+        "required": ["name"]
+      }
+    },
+    "general_plugins_phase": {
+      "allOf": [
+        {"$ref": "#/definitions/base_plugins_phase"},
+        {
+          "properties": {
+            "is_allowed_to_fail": {"type": "boolean"},
```
Comment (lcarva) [inline atomic_reactor/schemas/plugins.json]: Why is this not a property peer to `"name": {"type": "string"}` and `"args": {"type": "object"}` ?

Comment (mlangsdorf) [reply atomic_reactor/schemas/plugins.json]: I'm pretty sure it is; that's just how you write a schema that expands a ref.

Comment (lcarva) [reply atomic_reactor/schemas/plugins.json]: Oh I see, so you're expanding because `buildstep_plugins` do not allow these properties. Makes sense.

**Comment by lcarva** [comment]
Updated release notes draft:
> - `input_osv3` plugin validates`user_params` and `plugins_json` against a schema

**Pull request merged**